src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,122 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K,V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int)(DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (key == null) {
+            putForNullKey(value);
+            return;
+        }
+        int index = indexFor(hash(key), table.length);
+        for (Node<K,V> elem = table[index]; elem != null; elem = elem.next) {
+            Object k;
+            if (elem.hash == hash(key) && ((k = elem.key) == key || key.equals(k))) {
```

Comment:
```suggestion
            if (elem.hash == hash(key) && (key.equals(elem.key))) {
```
with k you made code longer

Reply:
done!